src/components/CIPipelineN/TaskTypeDetailComponent.tsx

```diff
@@ -0,0 +1,483 @@
+                const _formData = { ...formData }
+                _formData[activeStageName].steps[selectedTaskIndex].inlineStepDetail.script =
+                    '#!/bin/sh \nset -eo pipefail \n#set -v  ## uncomment this to debug the script \n' //default value for shell
+                setFormData(_formData)
+            }
+            return (
+                <>
+                    <CustomScript handleScriptChange={(e) => handleCustomChange(e, 'script')} />
+                    <div className="w-100 bcn-1 pt-7 br-8 flexbox h-32 pl-4 cn-9 fs-12">
+                        <Info className="path-info mt-2 mb-2 mr-8 ml-14 icon-dim-16" />
+                        <span>
+                            Your source code will be available at: <span className="fw-6">/devtroncd</span>
+                        </span>
+                    </div>
+                    <hr />
+                    <OutputDirectoryPath />
+                </>
+            )
+        }
+    }
+
+    function Option(_props) {
+        const { selectProps, data } = _props
+        selectProps.styles.option = getCustomOptionSelectionStyle({ direction: 'none', padding: '4px 10px' })
+        if (data.description) {
+            return (
+                <Tippy className="variable-description" arrow={false} placement="left" content={data.description}>
+                    <div className="flex left">
+                        <components.Option {..._props}>{_props.children}</components.Option>
+                    </div>
+                </Tippy>
+            )
+        } else {
+            return (
+                <div className="flex left">
+                    <components.Option {..._props}>{_props.children}</components.Option>
+                </div>
+            )
+        }
+    }
+
+    const handleContainerImageSelector = (selectedValue: OptionType) => {
+        setSelectedContainerImage(selectedValue)
+        const _formData = { ...formData }
+        _formData[activeStageName].steps[selectedTaskIndex].inlineStepDetail['containerImagePath'] = selectedValue.label
+        const _formErrorObject = { ...formDataErrorObj }
+        validateTask(
+            _formData[activeStageName].steps[selectedTaskIndex],
+            _formErrorObject[activeStageName].steps[selectedTaskIndex],
+        )
+        setFormDataErrorObj(_formErrorObject)
+        setFormData(_formData)
+    }
+
+    function handleCreatableBlur(e) {
+        if (e.target.value) {
+            setSelectedContainerImage({
+                label: e.target.value,
+                value: e.target.value,
+            })
+            const _formData = { ...formData }
+            _formData[activeStageName].steps[selectedTaskIndex].inlineStepDetail['containerImagePath'] = e.target.value
+            setFormData(_formData)
+        }
+    }
+
+    const handleKeyDown = (event) => {
+        switch (event.key) {
+            case 'Enter':
+            case 'Tab':
+                event.target.blur()
```

Comment:
I think it should be replaced with a normal `if statement` instead of a `switch`.